core/store/src/main/java/uk/gov/gchq/gaffer/store/operation/handler/AbstractSampleToSplitPointsHandler.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2017-2020 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.gaffer.store.operation.handler;
+
+import uk.gov.gchq.gaffer.commonutil.stream.Streams;
+import uk.gov.gchq.gaffer.operation.OperationException;
+import uk.gov.gchq.gaffer.operation.impl.SampleToSplitPoints;
+import uk.gov.gchq.gaffer.store.Context;
+import uk.gov.gchq.gaffer.store.Store;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public abstract class AbstractSampleToSplitPointsHandler <T, S extends Store> implements OutputOperationHandler<SampleToSplitPoints<T>, List<T>> {
+
+    @Override
+    public List<T> doOperation(final SampleToSplitPoints<T> operation, final Context context, final Store store) throws OperationException {
+
+        final S typedStore = (S) store;
+
+        validate(operation, typedStore);
+
+        final Integer numSplits = getNumSplits(operation, typedStore);
+        if (null == numSplits) {
```

Comment:
isNull(numSplits)

Reply:
For consistency I've kept the null checks as they were originally written prior to refactoring into this class.  Most of the code follows this pattern rather than using Objects.isNull.

Reply:
Federation is different them ;)